src/api/03-user-experience/09-google-cast.md

```diff
@@ -0,0 +1,247 @@
+    type: image/png
+  - sizes: 256x256
+    src: https://storage.googleapis.com/kino-assets/google-cast/artwork-256x256.png
+    type: image/png
+  - sizes: 384x384
+    src: https://storage.googleapis.com/kino-assets/google-cast/artwork-384x384.png
+    type: image/png
+  - sizes: 512x512
+    src: https://storage.googleapis.com/kino-assets/google-cast/artwork-512x512.png
+    type: image/png
+cast: true
+---
+
+## Introduction
+
+When one device controls media content playback on another device, we call it **casting**. These devices have to follow the same protocol in order to communicate. One of the most widely used casting protocols is Google Cast.
+
+## Basic concepts
+
+Before you can use Google Cast to stream your media to other devices, there are a few basic concepts you should understand first:
+
+* **Sender application**: A mobile or web application that uses Google Cast SDK to discover target devices on the network and to initialize and remotely control playback of media being cast to receiver.
+* **Receiver application**: A web application that runs on a Google Cast compatible device such as a Chromecast dongle or a Chromecast built-in TV or speaker and implements the media player UI and handles messages from the *Sender application*.
+* **Google Cast SDK**: Collection of API libraries and code samples to allow Android, iOS and web developers to implement sender and receiver applications.
+
+**Note:** Developing a custom Receiver application is optional in many use cases. Two prebuilt options are available:
+
+* [Default Media Web Receiver] – predefined UI and colors, but you don't need to [register] a custom receiver application.
+* [Styled Media Web Receiver] – ability to customize the UI and colors using CSS, but it is necessary to [register] a custom receiver application.
+
+## Using the Cast Application Framework
+
+The [Cast Application Framework] is the part of Google Cast SDK which exposes objects, methods and events necessary for building web sender applications.
+
+```html
+<script>
+// 1. When CAF initializes, the `__onGCastApiAvailable` is called.
+window.__onGCastApiAvailable = (isAvailable) => {
+  if (isAvailable) initCastApi();
+};
+
+function initCastApi() {
+  // 2. `CastContext` holds all global context for the CAF.
+  const context = cast.framework.CastContext.getInstance();
+
+  // 3. We use a default receiver app ID. If you want custom
+  //    styling or features, you must register your own app.
+  //
+  // @see https://developers.google.com/cast/docs/registration
+  const receiverId = chrome.cast.media.DEFAULT_MEDIA_RECEIVER_APP_ID;
+
+  context.setOptions({
+    receiverApplicationId: receiverId,
+  });
+
+  // 4. A `session` is a connection to a target device. Whenever its
+  //    state changes, we run the `castVideo` method.
+  context.addEventListener(
+    cast.framework.CastContextEventType.SESSION_STATE_CHANGED,
+    e => castVideo(e.sessionState),
+  );
+
+  const castVideo = asnyc (state) => {
```

Comment:
`s/asnyc/async/`

Reply:
Updated in 281290836b44cdc445dea0ef7410d1082e9d8506.